toqito/rand/random_states.py

```diff
@@ -0,0 +1,51 @@
+"""Module for generating random quantum states using Qiskit.
+
+Provides functionality to generate random quantum states with Qiskit's random_statevector function.
+Includes a function `random_states` to create a specified number of quantum states,
+each of a specified dimension.
+
+"""
+
+import numpy as np
+from qiskit.quantum_info.states.random import random_statevector
+
+
+def random_states(n: int, d: int) -> list[np.ndarray]:
+    """Generate a list of random quantum states.
+
+    Creates `n` random quantum states, each of dimension `d`. These states are generated using
+    Qiskit's `random_statevector` function, ensuring that they are valid quantum states
+    distributed according to the Haar measure.
+
+    Parameters
+    ----------
+    n : int
+        The number of random states to generate.
+    d : int
+        The dimension of each quantum state.
+
+    Returns
+    -------
+    list[numpy.ndarray]
+        A list of `n` numpy arrays. Each array is a d-dimensional quantum state represented
+        as a column vector.
```

Comment:
You do want to make sure you stay consistent with how these are listed in the docstrings of existing functions. 

For example,

https://github.com/vprusso/toqito/blob/5c1571ae483320ad40ce54d0a08f6a9bf31f2f4f/toqito/channel_ops/apply_channel.py#L93-L98

Also, make sure you add this addition to the `docs` as Vincent said in his comment.

Reply:
specifying parameter with :param tag is failing ``` ruff check .```  . is there any way to bypass it. I am getting this error if i remove argument description that i have included in previous pr and use :param tag.

`ruff check rand/ --fix
rand/random_states.py:7:5: D417 Missing argument descriptions in the docstring for `random_states`: `d`, `n`
Found 1 error.`

Reply:
Is your docstring following everything as expected? For example, Ruff is set to using `===` for section headings instead of `---`.

Pay attention to the full docstring I linked above and don't use an auto-formatter. FYI, we don't use `black`.

Reply:
i have pushed a change, pls review it and let me if it is ok or i need to make some more change.